OrbitClientModel/CaptureDeserializerTest.cpp

```diff
@@ -0,0 +1,408 @@
+         << serialized_header;
+
+  capture_deserializer::Load(stream, "not_existing_test_file", &listener, &cancellation_requested);
+
+  EXPECT_THAT(actual_error.message(), HasSubstr("1.51"));
+}
+
+TEST(CaptureDeserializer, LoadNoCaptureInfo) {
+  MockCaptureListener listener;
+  std::atomic<bool> cancellation_requested = false;
+  ErrorMessage actual_error;
+  EXPECT_CALL(listener, OnCaptureFailed).Times(1);
+  EXPECT_CALL(listener, OnCaptureStarted).Times(0);
+  EXPECT_CALL(listener, OnCaptureComplete).Times(0);
+  EXPECT_CALL(listener, OnCaptureCancelled).Times(0);
+  CaptureHeader header;
+  header.set_version(capture_deserializer::internal::kRequiredCaptureVersion);
+
+  std::string serialized_header;
+  header.SerializeToString(&serialized_header);
+  int32_t size_of_header = serialized_header.size();
+  std::stringstream stream;
+  stream << std::string(absl::bit_cast<char*>(&size_of_header), sizeof(size_of_header))
+         << serialized_header;
+
+  capture_deserializer::Load(stream, "not_existing_test_file", &listener, &cancellation_requested);
+}
+
+TEST(CaptureDeserializer, LoadCaptureInfoOnCaptureStarted) {
+  MockCaptureListener listener;
+  CaptureInfo capture_info;
+  capture_info.set_process_id(42);
+  capture_info.set_process_name("process");
+  FunctionInfo* selected_function = capture_info.add_selected_functions();
+  selected_function->set_name("foo");
+  selected_function->set_pretty_name("void foo()");
+  selected_function->set_address(21);
+  selected_function->set_size(12);
+  std::atomic<bool> cancellation_requested = false;
+  uint8_t empty_data = 0;
+  google::protobuf::io::CodedInputStream empty_stream(&empty_data, 0);
+
+  absl::flat_hash_map<uint64_t, orbit_client_protos::FunctionInfo> actual_selected_functions;
+  // There will be no call to OnCaptureStarted other then the one specified next.
+  EXPECT_CALL(listener, OnCaptureStarted(_, _, _, _, _)).Times(0);
+  EXPECT_CALL(listener, OnCaptureStarted(42, "process", NotNull(), _, IsEmpty()))
+      .Times(1)
+      .WillOnce(SaveArg<3>(&actual_selected_functions));
+  EXPECT_CALL(listener, OnCaptureComplete).Times(1);
+  EXPECT_CALL(listener, OnCaptureFailed).Times(0);
+  EXPECT_CALL(listener, OnCaptureCancelled).Times(0);
+
+  EXPECT_CALL(listener, OnAddressInfo).Times(0);
+  EXPECT_CALL(listener, OnThreadName).Times(0);
+
+  capture_deserializer::internal::LoadCaptureInfo(capture_info, &listener, &empty_stream,
+                                                  &cancellation_requested);
+
+  ASSERT_EQ(actual_selected_functions.size(), 1);
+  ASSERT_TRUE(actual_selected_functions.contains(selected_function->address()));
+  FunctionInfo actual_function_info = actual_selected_functions.at(selected_function->address());
+
+  EXPECT_EQ(actual_function_info.name(), selected_function->name());
+  EXPECT_EQ(actual_function_info.pretty_name(), selected_function->pretty_name());
+  EXPECT_EQ(actual_function_info.size(), selected_function->size());
```

Comment:
Would love the have those. However, they are auto generated by protos. GoogleTest will add a EqualsProto matcher at some point. But right now, I can't modify. I was about implementing my own custom matcher but thought it would be an overkill for this test.

Reply:
Ah, I see, didn't know that - makes sense, thanks for pointing that out.